pkg/servers/http2/agentGeneration.go

```diff
@@ -0,0 +1,123 @@
+package http2
```

Comment:
Probably need to move `agentGeneration.go` out of the `http2` package. I like the idea of the `utils` package or some other package like `modules` that doesn't conflict with the existing modules package used to read in and execute commands in a JSON file.

Reply:
hrm... I can approach it slightly differently, I was trying to avoid coupling the server with the agent template in order to make future transports easier to introduce, but because a lot of the agent code is modularised already I suspect that won't be as big of a problem as it could be...